src/metatensor/models/experimental/alchemical_model/train.py

```diff
@@ -90,12 +90,14 @@ def train(
 
     # Perform canonical checks on the datasets:
     logger.info("Checking datasets for consistency")
-    check_datasets(
-        train_datasets,
-        validation_datasets,
-        raise_incompatibility_error=continue_from is None,
-        # only error if we are not continuing
-    )
+    try:
```

Comment:
I moved the logic to raise a waring out of `check_datasets`. I think this cleaner.

Reply:
Looks good, thanks